### DISCUSSION TITLE
how to zip docker images in the correct way

### DISCUSSION THREAD

**Discussion by vrbgtrhr** [rancher/rke2#7306] [Q&A]
i have a lot of custom docker images that i want to use in the new rke2 cluster and i would like to package them all in one tar.zst like the images on github. what is the right procedure since there is also the implication of amd64/arm64 ?

**Comment by brandond** [comment]
`docker image save` does not handle multi-arch images properly. I would probably do something like:
```bash
ctr image pull --all-platforms --all-metadata image1 image2 image3
ctr image export --all-platforms --local images.tar image1 image2 image3
```

You can then compress the tarball using whatever tools you like. Note that when you export like this the layers are already gzip compressed though since containerd keeps the packed versions, so you probably won't get a TON of additional compression.

**Comment by vrbgtrhr** [comment]
ok the problem i have now is that with `sudo ctr image ls` i dont have any images but with  `sudo CRI_CONFIG_FILE=/var/lib/rancher/rke2/agent/etc/crictl.yaml /var/lib/rancher/rke2/bin/crictl image ls` i see all images, both the images loaded from `/var/lib/rancher/rke2/agent/images/` and downloaded form my local registry

**Reply by brandond** [reply]
It sounds like you have multiple container runtimes installed. Make sure that you don't have containerd, docker, or cri-o installed on your system. You should only have the containerd manage by RKE2.
